src/sample/deferredRendering/main.ts

```diff
@@ -0,0 +1,616 @@
+
+  if (canvasRef.current === null) return;
+  const context = canvasRef.current.getContext('gpupresent');
+
+  const aspect = Math.abs(canvasRef.current.width / canvasRef.current.height);
+
+  const swapChainFormat = 'bgra8unorm';
+  const swapChain = context.configureSwapChain({
+    device,
+    format: swapChainFormat,
+  });
+
+  // Create the model vertex buffer.
+  const kVertexStride = 8;
+  const vertexBuffer = device.createBuffer({
+    // position: vec3, normal: vec3, uv: vec2
+    size:
+      mesh.positions.length * kVertexStride * Float32Array.BYTES_PER_ELEMENT,
+    usage: GPUBufferUsage.VERTEX,
+    mappedAtCreation: true,
+  });
+  {
+    const mapping = new Float32Array(vertexBuffer.getMappedRange());
+    for (let i = 0; i < mesh.positions.length; ++i) {
+      mapping.set(mesh.positions[i], kVertexStride * i);
+      mapping.set(mesh.normals[i], kVertexStride * i + 3);
+      mapping.set(mesh.uvs[i], kVertexStride * i + 6);
+    }
+    vertexBuffer.unmap();
+  }
+
+  // Create the model index buffer.
+  const indexCount = mesh.triangles.length * 3;
+  const indexBuffer = device.createBuffer({
+    size: indexCount * Uint16Array.BYTES_PER_ELEMENT,
+    usage: GPUBufferUsage.INDEX,
+    mappedAtCreation: true,
+  });
+  {
+    const mapping = new Uint16Array(indexBuffer.getMappedRange());
+    for (let i = 0; i < mesh.triangles.length; ++i) {
+      mapping.set(mesh.triangles[i], 3 * i);
+    }
+    indexBuffer.unmap();
+  }
+
+  // GBuffer texture render targets
+  const gBufferTexture2DFloat = device.createTexture({
+    size: [canvasRef.current.width, canvasRef.current.height, 3],
+    usage: GPUTextureUsage.RENDER_ATTACHMENT | GPUTextureUsage.SAMPLED,
+    format: 'rgba32float',
+  });
+  const gBufferTextureAlbedo = device.createTexture({
+    size: [canvasRef.current.width, canvasRef.current.height, 1],
+    usage: GPUTextureUsage.RENDER_ATTACHMENT | GPUTextureUsage.SAMPLED,
+    format: 'bgra8unorm',
+  });
+  const gBufferTextureViews = [
+    gBufferTexture2DFloat.createView({ baseArrayLayer: 0, arrayLayerCount: 1 }),
+    gBufferTexture2DFloat.createView({ baseArrayLayer: 1, arrayLayerCount: 1 }),
+    gBufferTextureAlbedo.createView(),
+  ];
+
+  const vertexBuffers: Iterable<GPUVertexBufferLayout> = [
+    {
+      arrayStride: Float32Array.BYTES_PER_ELEMENT * 8,
+      attributes: [
+        {
+          // position
+          shaderLocation: 0,
+          offset: 0,
+          format: 'float32x3',
+        },
+        {
+          // normal
+          shaderLocation: 1,
+          offset: Float32Array.BYTES_PER_ELEMENT * 3,
+          format: 'float32x3',
+        },
+        {
+          // uv
+          shaderLocation: 2,
+          offset: Float32Array.BYTES_PER_ELEMENT * 6,
+          format: 'float32x2',
+        },
+      ],
+    },
+  ];
+
+  const primitive: GPUPrimitiveState = {
+    topology: 'triangle-list',
+    cullMode: 'back',
+  };
+
+  const WriteGBuffersPipeline = device.createRenderPipeline({
```

Comment:
nit: variables should be lowercase

```suggestion
  const writeGBuffersPipeline = device.createRenderPipeline({
```